File: src/main/java/com/irisine/jiris/compiler/assistpart/IrisTopExpressionApostrophe.java
package com.irisine.jiris.compiler.assistpart;

import com.irisine.jiris.compiler.IrisSyntaxUnit;
import com.irisine.jiris.compiler.expression.IrisExpression;

/**
 * Created by Huisama on 2017/4/17 0017.
 */
public class IrisTopExpressionApostrophe extends IrisSyntaxUnit {
    private IrisTopExpressionApostropheBase m_foreExpression = null;
    private IrisTopExpressionApostrophe m_postExpression = null;

    public IrisTopExpressionApostrophe(IrisTopExpressionApostropheBase foreExpression, IrisTopExpressionApostrophe postExpression) {
        m_foreExpression = foreExpression;
        m_postExpression = postExpression;
    }

    public IrisExpression ToDirectExpression(IrisExpression foreExpression) {
        return m_foreExpression.ToDirectExpression(foreExpression, m_postExpression);
    }
}

File: src/main/java/com/irisine/jiris/compiler/assistpart/IrisTopExpressionApostrophe.kt
package com.irisine.jiris.compiler.assistpart

import com.irisine.jiris.compiler.IrisSyntaxUnit
import com.irisine.jiris.compiler.expression.IrisExpression

/**
 * Created by Huisama on 2017/4/17 0017.
 */
class IrisTopExpressionApostrophe(val foreExpression: IrisTopExpressionApostropheBase?, val postExpression: IrisTopExpressionApostrophe?) : IrisSyntaxUnit() {

    fun ToDirectExpression(foreExpression: IrisExpression): IrisExpression {
        return this.foreExpression!!.ToDirectExpression(foreExpression, postExpression)
    }
}
